tests/test_etter_filter.py

```
@@ -0,0 +1,239 @@
+    def test_low_confidence_strict_raises(self):
+        upstream = MagicMock()
+        upstream.parse.return_value = SimpleNamespace(
+            spatial_relation="near",
+            reference_location="X",
+            buffer_config=None,
+            overall_confidence=0.4,
+        )
+        with patch.object(ef, "ETTER_AVAILABLE", True), \
+             patch.object(ef, "_UpstreamParser", return_value=upstream):
+            p = ef.EtterParser(llm=MagicMock(), confidence_threshold=0.6, strict_mode=True)
+            with pytest.raises(ef.EtterLowConfidenceError, match="0.40"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)